morf-core/src/main/java/org/alfasoftware/morf/upgrade/Upgrade.java

```diff
@@ -64,31 +61,34 @@
 public class Upgrade {
   private static final Log log = LogFactory.getLog(Upgrade.class);
 
-  private final UpgradePathFactory factory;
+  private final UpgradePathFactory upgradePathFactory;
   private final ConnectionResources connectionResources;
   private final UpgradeStatusTableService upgradeStatusTableService;
   private final ViewChangesDeploymentHelper viewChangesDeploymentHelper;
   private final ViewDeploymentValidator viewDeploymentValidator;
   private final GraphBasedUpgradeBuilderFactory graphBasedUpgradeBuilderFactory;
   private final DatabaseUpgradePathValidationService databaseUpgradePathValidationService;
+  private final UpgradeConfiguration upgradeConfiguration;
 
 
   public Upgrade(
       ConnectionResources connectionResources,
-      UpgradePathFactory factory,
+      UpgradePathFactory upgradePathFactory,
       UpgradeStatusTableService upgradeStatusTableService,
       ViewChangesDeploymentHelper viewChangesDeploymentHelper,
       ViewDeploymentValidator viewDeploymentValidator,
       GraphBasedUpgradeBuilderFactory graphBasedUpgradeBuilderFactory,
-      DatabaseUpgradePathValidationService databaseUpgradePathValidationService) {
+      DatabaseUpgradePathValidationService databaseUpgradePathValidationService,
+      UpgradeConfiguration upgradeConfiguration) {
     super();
     this.connectionResources = connectionResources;
-    this.factory = factory;
+    this.upgradePathFactory = upgradePathFactory;
     this.upgradeStatusTableService = upgradeStatusTableService;
     this.viewChangesDeploymentHelper = viewChangesDeploymentHelper;
     this.viewDeploymentValidator = viewDeploymentValidator;
     this.graphBasedUpgradeBuilderFactory = graphBasedUpgradeBuilderFactory;
     this.databaseUpgradePathValidationService = databaseUpgradePathValidationService;
+    this.upgradeConfiguration = upgradeConfiguration;
   }

```

Comment:
Note: This is a deliberate backwards-incompatible change.
Long term plan is to make this constructor fully private, and move all users to use the Factory instead, reducing the number of access points to this class.